lib/hooks/useEvent.lua

```diff
@@ -92,7 +94,18 @@ local function useEvent(instance, event): () -> (number, ...any)
 		storage.queue = queue
 		storage.event = event
 
-		local connection = event:Connect(function(...)
+		-- Get the 'Connect' method through duck typing, since sometimes developers
+		-- may have custom events that they want to connect through this function:
+		local connectMethod 
+
+		for _, method in EVENT_CONNECT_METHODS do
+			if type(event[method]) == "function" then
```

Comment:
Indexing directly into Roblox instances when the key doesn't exist errors. Do you think we should account for that case here explicitly by checking if the object is a userdata or a table?